src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1173,6 +1190,154 @@ ChooseGenerator::ChooseGenerator(const Node& node, GeneratorArgs generatorArgs)
     }
 }
 
+/**
+ * @private
+ * @param exception exception received from Generator factory (e.g. ::intGenerator()) invocation.
+ * @return if the exception is a result of the generator being unknown to the factory. For
+ * example, a ^RandomDate node being passed to the intGenerator().
+ */
+bool isUnknownParser(const InvalidValueGeneratorSyntax& exception) {
+    auto what = std::string{exception.what()};
+    return what.find("Unknown parser") != std::string::npos;
```

Comment:
we could make a separate `UnknownParserException` if that would help? Relying on exception messages for logic has bitten me a number of times in the past (trying to make exceptions easier to read has caused bugs). If that doesn't seem worthwhile or like it will work plz add a comment wherever it adds the "Unknown parser" string letting future maintainers know about `isUnknownParser`. (Obv same to the format one below)